### ISSUE TITLE
Introduce micro-optimization of (long) Unsafe.ARRAY_BYTE_BASE_OFFSET in UnsafeMemory

### ISSUE THREAD

**Issue by minborg** [OpenHFT/Chronicle-Core#336]

**Comment by minborg** [comment]
Reding from `Unsafe.ARRAY_BYTE_BASE_OFFSET`:

```
  public static void unsafePutLong(byte[], int, long);
    Code:
       0: getstatic     #11                 // Field UNSAFE:Lsun/misc/Unsafe;
       3: aload_0
       4: getstatic     #32                 // Field sun/misc/Unsafe.ARRAY_BYTE_BASE_OFFSET:I
       7: i2l
       8: iload_1
       9: i2l
      10: ladd
      11: lload_2
      12: invokevirtual #42                 // Method sun/misc/Unsafe.putLong:(Ljava/lang/Object;JJ)V
      15: return
```

Reading from a local variable:

```
  public static void unsafePutLong(byte[], int, long);
    Code:
       0: getstatic     #11                 // Field UNSAFE:Lsun/misc/Unsafe;
       3: aload_0
       4: getstatic     #42                 // Field ARRAY_BYTE_BASE_OFFSET:J
       7: iload_1
       8: i2l
       9: ladd
      10: lload_2
      11: invokevirtual #43                 // Method sun/misc/Unsafe.putLong:(Ljava/lang/Object;JJ)V
      14: return
```

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.22ea23](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.22ea23), [BOM-2.22ea86](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22ea86)
